keras_contrib/applications/densenet.py

```diff
@@ -258,6 +368,38 @@ def __dense_block(x, nb_layers, nb_filter, growth_rate, bottleneck=False, dropou
     return x, nb_filter
 
 
+def __transition_up_block(ip, nb_filters, type='upsampling', output_shape=None, weight_decay=1E-4):
+    ''' SubpixelConvolutional Upscaling (factor = 2)
+
+    Args:
+        ip: keras tensor
+        nb_filters: number of layers
+        type: can be 'upsampling', 'subpixel', 'deconv', or 'atrous'. Determines type of upsampling performed
+        output_shape: required if type = 'deconv'. Output shape of tensor
+        weight_decay: weight decay factor
+
+    Returns: keras tensor, after applying batch_norm, relu-conv, dropout, maxpool
+    '''
+
+    if type == 'upsampling':
+        x = UpSampling2D()(ip)
+    elif type == 'subpixel':
+        x = Convolution2D(nb_filters, 3, 3, activation="relu", border_mode='same', W_regularizer=l2(weight_decay),
+                          bias=False)(ip)
+        x = SubPixelUpscaling(r=2, channels=int(nb_filters // 4))(x)
+        x = Convolution2D(nb_filters, 3, 3, activation="relu", border_mode='same', W_regularizer=l2(weight_decay),
+                          bias=False)(x)
+    elif type == 'atrous':
+        # waiting on https://github.com/fchollet/keras/issues/4018
+        x = AtrousConvolution2D(nb_filters, 3, 3, activation="relu", W_regularizer=l2(weight_decay),
```

Comment:
okay that sounds like a good for now!

> How does that even work?

See Multi-Scale Context Aggregation by Dilated Convolutions
https://arxiv.org/abs/1511.07122

Also some of the materials myself and patyork links to in https://github.com/fchollet/keras/issues/4018 have info specific to this.